week-1-ts-ecosystem/w01e03-object-oriented-structure/models/Product.ts

```diff
@@ -0,0 +1,88 @@
+export type ProductType = "BUY_NOW" | "ON_AUCTION" | "FOR_FREE";
+
+export interface UpdateProductDTO {
+  name?: string;
+  quantity?: number;
+  price?: number;
+}
+
+export class Product {
+  #id: number;
+  #name: string;
+  #quantity: number;
+  #price: number;
+  #type: ProductType;
+
+  constructor(
+    id: number,
+    name: string,
+    quantity: number,
+    price: number,
+    type: ProductType
+  ) {
+    this.#id = id;
```

Comment:
Te początkowe ustawienia elementów nie powinny bezpośrednio uderzać w prywatne pola `#`
ponieważ w `setterach` ustaliłeś pewną walidację

W tym układzie np. jeśli zrobię tak:
```typescript
new Product(1,'',-100,-100,"ON_AUCTION")
```

to to przejdzie - a nie powinno.
Dzieje się tak dlatego że `setters` nie odpalą się....